Added minimal support for cluster logs collection API

cosimo opened the pull request:
https://docs.couchbase.com/server/7.2/rest-api/rest-manage-log-collection.html

cosimo-kahoot committed:
Added start_logs_collection() method

This is to automate the cluster logs collection operation.
Had to add a `get_cluster_nodes()` method too, as the documented `nodes='*'`
parameter for startLogsCollection does not seem to work.

The error message that I get back is always the same, `"nodes":{"needs
to be a list of nodes"}` or something along those lines.
I tried a lot of different combinations, but none worked.

Each node data structure returned by `get_cluster_nodes()` contains a `otpNode`
attribute which is what we need to provide to `start_logs_collection()`.

cosimo-kahoot committed:
Added tests for get_cluster_nodes(), start_logs_collection()

sveniu commented on couchbase_cluster_admin/cluster.py:
```diff
@@ -85,6 +85,25 @@ def enable_services(self):
+            raise GetClusterNodesException(resp.text)
+
+        cluster_info = resp.json()
+        nodes = cluster_info.get("nodes", [])
```
When would there be no `nodes` attribute? If there is no such attr, is it suitable to return an empty list, as opposed to raising a specific exception? I'm not sure myself.

cosimo replied:
I will instead use the existing `known_nodes`.

sveniu commented on couchbase_cluster_admin/cluster.py:
```diff
@@ -85,6 +85,25 @@ def enable_services(self):
         if resp.status_code != 200:
             raise Exception(f"Failed to enable services: {resp.text}")
 
+    def get_cluster_nodes(self):
```
This copies most of the self.pool_info() property. It could indeed return `self.pool_info["nodes"]` to make it simpler.

There's also a vaguely named self.known_nodes() property that returns otpNode of each node.

cosimo replied:
Thanks, I had not seen those. Will rework this then.

cosimo-kahoot committed:
Review: removed get_cluster_nodes() method

The same functionality is already provided by the `pool_info` and `known_nodes`
attributes. There was no need for a new method.

sveniu commented on couchbase_cluster_admin/exceptions.py:
```diff
@@ -26,10 +26,18 @@ class DeleteAlternateAddressException(Exception):
     pass
 
 
+class GetClusterNodesException(Exception):
+    pass
+
+
```
This can go too?

cosimo replied:
Yes :-/

cosimo-kahoot committed:
Review: removed get_cluster_nodes() related exception class

cosimo-kahoot committed:
Fixed failing tests due to missing `docker-compose` command

Github removed `docker-compose` from their runner images on July 9th, 2024.
Ref: https://github.blog/changelog/2024-04-10-github-hosted-runner-images-deprecation-notice-docker-compose-v1/

> Docker Compose v1 has been deprecated as of July 2023. All customers utilizing Compose v1 on GitHub-hosted runners are encouraged to migrate to Compose v2. Per GitHub’s support policy we will remove this tool from our GitHub managed runner images effective July 9, 2024.
>
> To avoid breaking changes, customers will need to update their Actions workflow files from using docker-compose to docker compose. After July 9, workflows will begin to fail that are using the previous syntax. Customers are advised to review the migration instructions to ensure they are making all the changes required.

cosimo-kahoot committed:
Removed unused json import
